OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -220,14 +220,18 @@ Actor ChooseTarget(Actor self, Stance attackStances, WDist range, bool allowMove
 
 			return null;
 		}
+	}
 
-		bool PreventsAutoTarget(Actor attacker, Actor target)
+	public static class TargetExts
+	{
+		public static bool MethodIsValidFor(this Target target, Actor targeter)
 		{
-			foreach (var pat in target.TraitsImplementing<IPreventsAutoTarget>())
-				if (pat.PreventsAutoTarget(target, attacker))
-					return true;
+			return !(target.Actor != null && target.TargetMethod == TargetMethod.Automatic && target.Actor.PreventsAutoTarget(targeter));
+		}
 
-			return false;
+		public static bool AndMethodIsValidFor(this Target target, Actor targeter)
```

Comment:
This name derives from the names referred to from the method: It conjugates the results of the method `IsValidFor` and the method `MethodIsValidFor`.
Another option that came to mind is changing the name of the existing method `Target.IsValidFor` to something which signifies that the `TargetMethod` introduced with this change is ignored and then use the name `IsValidFor` for the method you referred to.